helper.php

```diff
@@ -550,10 +552,12 @@ private function adapt_links(&$ins, $page, $included_pages = null) {
             // adjust links with image titles
             if (strpos($ins[$i][0], 'link') !== false && isset($ins[$i][1][1]) && is_array($ins[$i][1][1]) && $ins[$i][1][1]['type'] == 'internalmedia') {
                 // resolve relative ids, but without cleaning in order to preserve the name
-                $media_id = resolve_id($ns, $ins[$i][1][1]['src']);
+                $resolver = new PageResolver($ns);
+                $ins[$i][1][1]['src'] = $resolver->resolveId($ins[$i][1][1]['src']);
                 // make sure that after resolving the link again it will be the same link
                 if ($media_id[0] != ':') $media_id = ':'.$media_id;
                 $ins[$i][1][1]['src'] = $media_id;
+                unset($resolver);
```

Comment:
Not wrong but I don't think this pattern is used much in DokuWiki. PHP will clean up variables and objects going out of scope without manual intervention.

Reply:
That's my background in C/C++ kicking in.